test/jdk/java/nio/file/Path/Extensions.java

```diff
@@ -42,38 +42,38 @@ public class Extensions {
     @DataProvider
     static Object[][] getProvider() {
         Object[][] pairs = {
-            {"",               null},
-            {".",              null},
-            {"..",             ""},
-            {"...",            ""},
-            {"....",           ""},
-            {".....",          ""},
-            {"aa",             null},
-            {"a.",             ""},
-            {".a",             null},
-            {"..a",            "a"},
-            {"...a",           "a"},
-            {"....a",          "a"},
-            {".a.b",           "b"},
-            {"...a.b",         "b"},
-            {"...a.b.",        ""},
-            {"..foo",          "foo"},
-            {"foo.",           ""},
-            {"test.",          ""},
-            {"test..",         ""},
-            {"test...",        ""},
-            {"test.rb",        "rb"},
-            {"a/b/d/test.rb" , "rb"},
-            {".a/b/d/test.rb", "rb"},
-            {"test",           null},
-            {".profile",       null},
-            {".profile.sh",    "sh"},
-            {"foo.tar.gz",     "gz"},
-            {"foo.bar.",       ""},
-            {"archive.zip",    "zip"},
-            {"compress.gzip",  "gzip"},
-            {"waitwhat.&$!#%", "&$!#%"},
-            {"6.283185307",    "283185307"}
+            {"",               ""},
+            {".",              ""},
+            {"..",             "."},
+            {"...",            "."},
+            {"....",           "."},
+            {".....",          "."},
+            {"aa",             ""},
+            {"a.",             "."},
+            {".a",             ""},
+            {"..a",            ".a"},
+            {"...a",           ".a"},
+            {"....a",          ".a"},
+            {".a.b",           ".b"},
+            {"...a.b",         ".b"},
+            {"...a.b.",        "."},
+            {"..foo",          ".foo"},
+            {"foo.",           "."},
+            {"test.",          "."},
+            {"test..",         "."},
+            {"test...",        "."},
+            {"test.rb",        ".rb"},
+            {"a/b/d/test.rb" , ".rb"},
+            {".a/b/d/test.rb", ".rb"},
+            {"test",           ""},
+            {".profile",       ""},
+            {".profile.sh",    ".sh"},
+            {"foo.tar.gz",     ".gz"},
+            {"foo.bar.",       "."},
+            {"archive.zip",    ".zip"},
+            {"compress.gzip",  ".gzip"},
+            {"waitwhat.&$!#%", ".&$!#%"},
+            {"6.283185307",    ".283185307"}
```

Comment:
For completeness you might also want to include:
```
a..b
a..b.c
a.b..c
.a..b.
.a..b
.a..b.c
.a.b..c
.a..b.
```

Reply:
Added in af0980652e88297166251d44dd7d0ae79b8e6633.